test/tests/autotranslation.js

```diff
@@ -18,11 +18,16 @@ var BeakerPageObject = require('./beaker.po.js');
 var path = require('path');
 
 describe('autotranslation', function() {
-  beforeEach(function() {
+  beforeEach(function(done) {
     jasmine.DEFAULT_TIMEOUT_INTERVAL = 120000; // Slow initialization cells, CI box might need a long time to initialize these
     beakerPO = new BeakerPageObject();
-    browser.get(beakerPO.baseURL);
-    browser.waitForAngular();
+    browser.get(beakerPO.baseURL).then(
+        function(){done();},
+        function(reason){ console.log('error occurred while browser load page : ' + reason);}
+    );
+    browser.waitForAngular().then(function() {
+      done();
```

Comment:
Didn't know this that Protractor does waitForAngular automatically, thanks.

If that's the case, can we just remove the waitForAngular call completely? I don't believe in second chances in the code. I would hope that just making that beforeEach block async would help us to avoid this error.

I wouldn't go with the timeouts, we already have far too many of those. If you think that we still need the "second chance", I would definitely go with the second example of that promise code:

```javascript
browser.get(beakerPO.baseURL)
  .then(done, browser.waitForAngular)
  .then(done);
```
But I would much rather first try it without waitForAngular and just let the beforeEach wait on browser.get to complete.